### DISCUSSION TITLE
EVALUATE all columns with measures

### DISCUSSION THREAD

**Discussion by matussumilas** [DaxStudio/DaxStudio#634] [Q&A] [answered]
Hello, would be there  a way to get all columns with measures with one command? 
I am using DAX in power pivot excel where I calculate couple of measures and I wanted to retrieve all data back to sheet for users. 
In my example I write EVALUATE GOLD_Q but it shows only columns in the table but not measures.

I know there could be other options where I just build dax query with all columns and measures but in my case users are able to change column in GOLD_Q so I would need to build a VBA macro which always build dax query with current columns.

I was just looking if it is possible to write something like
EVALUATE 
GOLD_Q
measure1
measure2

![image](https://user-images.githubusercontent.com/58041280/119977870-35f47400-bfb9-11eb-9dd2-12dc1fe77e66.png)

**Comment by dgosbell** [comment, marked as answer]
You can use the ADDCOLUMNS function to do this as it can take any form of expression including measures to include additional columns in the output

eg

```
EVALUATE
ADDCOLUMNS(
GOLD_Q,
"measure1", [measure1],
"measure2", [measure2]
)
```

**Reply by matussumilas** [reply]
Thank you!
